frontend/src/app/(routes)/staking/components/StakingOverview.tsx

```diff
@@ -0,0 +1,119 @@
+'use client';
+
+import React, { useEffect } from 'react';
+import './../staking.css';
+import { useAppDispatch, useAppSelector } from '@/custom-hooks/StateHooks';
+import { RootState } from '@/store/store';
+import {
+  getAllValidators,
+  getDelegations,
+  getUnbonding,
+} from '@/store/features/staking/stakeSlice';
+import ChainDelegations from './ChainDelegations';
+import ChainUnbondings from './ChainUnbondings';
+import { getDelegatorTotalRewards } from '@/store/features/distribution/distributionSlice';
+
+const StakingOverview = () => {
+  const dispatch = useAppDispatch();
+  const networks = useAppSelector((state: RootState) => state.wallet.networks);
+  const nameToChainIDs: Record<string, string> = useAppSelector(
+    (state: RootState) => state.wallet.nameToChainIDs
+  );
+  const chainIDs = Object.keys(nameToChainIDs).map(
+    (chainName) => nameToChainIDs[chainName]
+  );
+
+  const stakingData = useAppSelector(
+    (state: RootState) => state.staking.chains
+  );
+  const rewardsData = useAppSelector(
+    (state: RootState) => state.distribution.chains
+  );
+  useEffect(() => {
+    if (chainIDs) {
+      chainIDs.forEach((chainID) => {
+        const allChainInfo = networks[chainID];
```

Comment:
I see lot of the places you are writing the same logic, so write some utility function to get all these values. pass chain Id.
```suggestion
        const {chainInfo, address, baseURL, currency} = getBasicChainInfo(chainID);
```

and you can rewrite logic like this
```suggestion
        const { network, walletInfo } = networks[chainID] || {};
const { rest } = network?.config || {};
const { bech32Address } = walletInfo || {};
const currency = network?.config?.currencies?.[0];
```